domains/certificates/Query.API/API/Query.API/Projections/CertificatesByOwnerProjection.cs

```diff
@@ -1,119 +1,153 @@
-        view.Certificates[@event.CertificateId] = new CertificateView
+        view ??= GetCertificatesByOwnerView(operations, productionCertificateIssued.CertificateId);
+
+        view.Certificates[productionCertificateIssued.CertificateId].Status = CertificateStatus.Issued;
+        return view;
+    }
+
+    private static CertificatesByOwnerView CreateCertificatesByOwnerView(IDocumentOperations operations,
+        ProductionCertificateCreated productionCertificateCreated)
+    {
+        var view = operations.Load<CertificatesByOwnerView>(productionCertificateCreated.MeteringPointOwner);
+        var certificateView = new CertificateView
         {
-            CertificateId = @event.CertificateId,
-            DateFrom = @event.Period.DateFrom,
-            DateTo = @event.Period.DateTo,
-            Quantity = @event.ShieldedQuantity.Value,
-            GSRN = @event.ShieldedGSRN.Value,
-            GridArea = @event.GridArea,
-            TechCode = @event.Technology.TechCode,
-            FuelCode = @event.Technology.FuelCode,
+            CertificateId = productionCertificateCreated.CertificateId,
+            DateFrom = productionCertificateCreated.Period.DateFrom,
+            DateTo = productionCertificateCreated.Period.DateTo,
+            Quantity = productionCertificateCreated.ShieldedQuantity.Value,
+            GSRN = productionCertificateCreated.ShieldedGSRN.Value,
+            GridArea = productionCertificateCreated.GridArea,
+            TechCode = productionCertificateCreated.Technology.TechCode,
+            FuelCode = productionCertificateCreated.Technology.FuelCode,
             Status = CertificateStatus.Creating
         };
-    }
 
-    public void Apply(ProductionCertificateIssued @event, CertificatesByOwnerView view)
-        => view.Certificates[@event.CertificateId].Status = CertificateStatus.Issued;
+        if (view == null)
+        {
+            return new CertificatesByOwnerView
+            {
+                Owner = productionCertificateCreated.MeteringPointOwner,
+                Certificates =
+                {
+                    [productionCertificateCreated.CertificateId] = certificateView
+                }
+            };
+        }
 
-    public void Apply(ProductionCertificateRejected @event, CertificatesByOwnerView view)
-        => view.Certificates[@event.CertificateId].Status = CertificateStatus.Rejected;
-}
+        view.Certificates.Add(productionCertificateCreated.CertificateId, certificateView);
+        return view;
+    }
 
-public class CertificatesByOwnerView
-{
-    [Identity]
-    public string Owner { get; set; } = "";
+    private static CertificatesByOwnerView ApplyProductionCertificateTransferred(IDocumentOperations operations,
+        ProductionCertificateTransferred productionCertificateTransferred)
+    {
+        var source = operations.Load<CertificatesByOwnerView>(productionCertificateTransferred.Source)
+                     ?? throw new CertificateDomainException(
```

Comment:
I think we should use a different exception type here as `CertificateDomainException` is thought to be used dealing with "illegal" actions on a certificate, not that a projection is failing. I suggest that we create a new type called e.g. `ProjectionException`.